zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/ProcessElementProperties.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.processing.bpmn;
+
+import static io.camunda.zeebe.util.buffer.BufferUtil.bufferAsString;
+
+import io.camunda.zeebe.protocol.impl.record.value.usertask.UserTaskRecord;
+
+public final class ProcessElementProperties {
```

Comment:
❓ What's the reason for introducing this new properties class? It seems to cover the same data as the `BpmnElementContext` and has a similar name.

Reply:
Thank you, @korthout, for pointing this out! 🙏
I wasn't entirely happy with this solution either, and your comment prompted me to rethink it.
Initially, I introduced `ProcessElementProperties` because it wasn't obvious that I could simply initialize `BpmnElementContext` inside `UserTaskProcessor` at the places where I needed it :sweat_smile:

After your feedback, I took a closer look and found a better approach:
now I'm building the `BpmnElementContext` directly inside the `UserTaskProcessor` from user task `ElementInstance`, which made the `ProcessElementProperties` class redundant.
So, removed it as part of this [commit](https://github.com/camunda/camunda/pull/22576/commits/b0fc3e250481ccba37e8467db63a063958e186ab).